### ISSUE TITLE
Training does not converge when using batch parameter

### ISSUE THREAD

**Issue by thiagoanglervision** [ultralytics/ultralytics#21419]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

When I change the batch parameter to 0.7 or even -1 for automatic, the training does not converge anymore.
If I comment it, training converges as normal.

### Environment

Ultralytics 8.3.167 🚀 Python-3.12.11 torch-2.5.1+cu124 CUDA:0 (NVIDIA RTX 5000 Ada Generation, 32362MiB)
Setup complete ✅ (48 CPUs, 62.3 GB RAM, 829.0/915.3 GB disk)

OS                  Linux-6.8.0-60-generic-x86_64-with-glibc2.35
Environment         Linux
Python              3.12.11
Install             git
Path                /home/tmedeiros/Code/AquaModelTraining/.venv/lib/python3.12/site-packages/ultralytics
RAM                 62.30 GB
Disk                829.0/915.3 GB
CPU                 AMD Ryzen Threadripper 7960X 24-Cores
CPU count           48
GPU                 NVIDIA RTX 5000 Ada Generation, 32362MiB
GPU count           1
CUDA                12.4

numpy               ✅ 1.26.4>=1.23.0
matplotlib          ✅ 3.10.0>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 11.1.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.15.0>=1.4.1
torch               ✅ 2.5.1>=1.8.0
torch               ✅ 2.5.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.20.1>=0.9.0
tqdm                ✅ 4.67.1>=4.64.0
psutil              ✅ 6.1.1
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.3>=1.1.4
ultralytics-thop    ✅ 2.0.14>=2.0.0

### Minimal Reproducible Example

from ultralytics import YOLO

size_training = 768

model = YOLO("yolo11n.pt")

results = model.train(data="./dataset/data.yaml"
                          , epochs=400
                          , imgsz=size_training
                          , patience=100
                          , workers=4
                          , deterministic=False
                          , hsv_h=0.015
                          , hsv_s=0.7
                          , hsv_v=0.4
                          , degrees=5.0
                          , scale=0.5
                          , mosaic=0.5
                          , mixup=0.2
                          , fliplr=0.5
                          , batch=0.7
                          )

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @thiagoanglervision, thank you for your detailed report and for using Ultralytics 🚀! This is an automated response to help you get started—an Ultralytics engineer will also assist you soon.

We recommend checking our [Docs](https://docs.ultralytics.com/) for guidance on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage, especially regarding training parameters like `batch`. Many common training questions are addressed there.

Since this is a 🐛 Bug Report, thank you for including a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/)—that really helps us debug!

To help us further investigate:
1. Please verify that you are using the latest version of the `ultralytics` package, as issues may have been resolved in newer releases. Upgrade with:
    ```bash
    pip install -U ultralytics
    ```
2. Make sure all dependencies listed in the [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) are up to date and that your Python environment is [**Python>=3.8**](https://www.python.org/) with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).

For your specific environment, you can also try running YOLO in any of these up-to-date, verified platforms (with all dependencies preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

You can also connect with the Ultralytics community for support and discussion:
- Real-time chat: [Discord](https://discord.com/invite/ultralytics) 🎧
- In-depth Q&A: [Discourse](https://community.ultralytics.com/)
- Community threads: [Subreddit](https://www.reddit.com/r/Ultralytics/)

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If the badge above is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI covers all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu.

Thank you for helping us make Ultralytics better! An engineer will review your issue and follow up soon.

**Comment by Y-T-G** [comment]
Can you post the `results.jpg` from both trainings?

**Comment by thiagoanglervision** [comment]
I found the culprit.

I was importing **export_yolo11** from [DeepStream-Yolo](https://github.com/marcoslucianops/DeepStream-Yolo) so I could export the the best result to an onnx file. I was following the instructions from [Ultralytics YOLO11 on NVIDIA Jetson using DeepStream SDK and TensorRT](https://docs.ultralytics.com/guides/deepstream-nvidia-jetson/#deepstream-configuration-for-yolo11)

As soon as I remove that import, the training converges.

Is there a way to have both working together?

**Comment by Y-T-G** [comment]
You should move the `import` to be after the training code.

**Comment by thiagoanglervision** [comment]
It is a constant training script, where I train different size models and I also shuffle my dataset.
Because of this, even after I do that, it will import and then go back to training as in the snippet below.
On the second run, after exporting, it will not converge again.

```
from ultralytics import YOLO

def train_model(model):

    results = model.train(data="./dataset/data.yaml"
                          , epochs=400
                          , imgsz=size_training
                          , patience=100
                          # , workers=4
                          , deterministic=False
                          , hsv_h=0.015
                          , hsv_s=0.7
                          , hsv_v=0.4
                          , degrees=5.0
                          , scale=0.5
                          , mosaic=0.5
                          , mixup=0.2
                          , fliplr=0.5
                          , batch=0.7
                          )

    model_path = str(results.save_dir)

    return model_path

training_model_types = ["yolo11n.pt", "yolo11s.pt", "yolo11m.pt", "yolo11l.pt", "yolo11x.pt"]

while True:
    for model_type in training_model_types:
        model = YOLO(model_type)

        new_model_path = train_model(model)
        weights=new_model_path + "/weights/best.pt"

       import argparse
       import export_yolo11

       args = argparse.Namespace()

       args.weights = weights  # REQUIRED: Set the path to your weights file
       args.size = [size_training]  # Default is [640]
       args.opset = opset  # Default is 12
       args.simplify = simplify  # Set to True if you want to simplify the model
       args.dynamic = dynamic  # Set to True for dynamic batch size
       args.batch = batch  # Default is 1, only used if dynamic is False


       print("Starting the YOLO to ONNX conversion")
       export_yolo11.main(args)
       print("Conversion finished.")
```

**Comment by Y-T-G** [comment]
You can use subprocess to run the export script instead of importing.
